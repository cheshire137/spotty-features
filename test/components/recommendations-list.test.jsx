import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import RecommendationsList from '../../src/components/recommendations-list.jsx'

import RecommendationsResponse from '../fixtures/spotify/recommendations'
import TrackSearchResponse from '../fixtures/spotify/track-search'

const recommendation = RecommendationsResponse.tracks[0]

const track = TrackSearchResponse.tracks.items[0]
const trackSeed = {
  id: track.id,
  name: track.name,
  url: track.external_urls.spotify,
  artists: track.artists.map(artist => artist.name),
  album: track.album.name,
  image: track.album.images[2].url,
  albumUrl: track.album.external_urls.spotify
}

function props(opts) {
  return {
    unauthorized: () => {},
    changeAudioFeatures: opts.changeAudioFeatures,
    recommendations: [
      {
        album: recommendation.album.name,
        albumUrl: recommendation.album.external_urls.spotify,
        artists: recommendation.artists.map(a => a.name),
        id: recommendation.id,
        image: recommendation.album.images[2].url,
        name: recommendation.name,
        spotifyUri: recommendation.uri,
        url: recommendation.external_urls.spotify
      }
    ],
    token: '123abc',
    seedType: 'track',
    seed: trackSeed
  }
}

describe('RecommendationsList', () => {
  let component = null
  let wantToChangeAudioFeatures = false

  const changeAudioFeatures = () => {
    wantToChangeAudioFeatures = true
  }

  beforeEach(() => {
    const opts = { changeAudioFeatures }
    component = <RecommendationsList {...props(opts)} />
  })

  afterEach(() => {
    wantToChangeAudioFeatures = false
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('can go back to change audio features', () => {
    const button = shallow(component).find('.button.is-link')
    button.simulate('click')
    expect(wantToChangeAudioFeatures).toBe(true)
  })
})
