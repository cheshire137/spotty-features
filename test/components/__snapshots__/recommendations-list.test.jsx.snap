exports[`RecommendationsList matches snapshot 1`] = `
<div
  className="content">
  <p>
    <button
      className="button is-link"
      onClick={[Function]}
      type="button">
      ‚Üê Change filters
    </button>
  </p>
  <h3
    className="title is-3 song-recs-title">
    Step 3: Save playlist (
    <span>
      1 song
    </span>
    )
  </h3>
  <form
    className="playlist-form"
    onSubmit={[Function]}>
    <div
      className="control has-addons">
      <input
        className="input is-expanded is-large"
        onChange={[Function]}
        type="text"
        value="Like SCREAM by Grimes" />
      <button
        className="button is-large is-primary"
        type="submit">
        Create Playlist
      </button>
    </div>
  </form>
  <ul
    className="recommendations-list">
    <li>
      <div
        className="columns">
        <div
          className="column track-image-column">
          <a
            className="track-link"
            href="https://open.spotify.com/album/3o5EnVZNJXtfPV8tCoagjI"
            rel="noopener noreferrer"
            target="_blank">
            <img
              className="track-image"
              src="https://i.scdn.co/image/d4a10ce84788eb82b53f72d1dee2fca4fa8cbcf0" />
          </a>
        </div>
        <div
          className="column recommended-track-details">
          <a
            className="track-link"
            href="https://open.spotify.com/track/6DLJQz0hav4l4VfEjJze5T"
            rel="noopener noreferrer"
            target="_blank">
            <span
              className="track-name">
              The Way Things Are
            </span>
            <span>
              by
            </span>
            <span
              className="track-artists">
              Fiona Apple
            </span>
          </a>
          <a
            className="track-link"
            href="https://open.spotify.com/album/3o5EnVZNJXtfPV8tCoagjI"
            rel="noopener noreferrer"
            target="_blank">
            <span
              className="track-album">
              When The Pawn...
            </span>
          </a>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
